Makefile

```diff
@@ -0,0 +1,160 @@
+ROOT_DIR := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+export LLC := llc
+export CLANG := clang
+export OBJCOPY := llvm-objcopy
+
+CARGO := cargo
+RELEASE_VERSION ?= $(shell $(CARGO) metadata --no-deps --format-version=1 | jq -r '.packages | .[] | select(.name == "retis") | .version')
+RELEASE_NAME ?= $(shell $(CARGO) metadata --no-deps --format-version=1 | jq '.packages | .[] | select(.name=="retis") | .metadata.misc.release_name')
+
+PRINT = echo
+
+VERBOSITY := $(filter 1,$(V))$(filter 2,$(V))
+
+ifeq ($(VERBOSITY),)
+    Q=@
+    MAKE += -s
+    CARGO += -q
+define out_console
+    $(PRINT) -e "[$(1)]\t$(2)"
+endef
+
+.SILENT:
+else
+    ifeq ($(VERBOSITY),2)
+        CARGO_CMD_OPTS += --verbose
```

Comment:
The same goes for release vs build + CARGO_OPTS.
That was an attempt to humor V=1,2 request :)

If the suggestion is still `V=1, V=2, V=12`, I think that any other further behavior finds better room in follow-ups otherwise the time for the PR stretches unreasonably adding more and more features/customizations (given in the old way of building we had no features like that).
Not sure we want to drop this V={1,2} part, though :D

Reply:
> The same goes for release vs build + CARGO_OPTS.

A suggestion below would solve that ;)

> If the suggestion is still `V=1, V=2, V=12`, I think that any other further behavior finds better room in follow-ups otherwise the time for the PR stretches unreasonably adding more and more features/customizations (given in the old way of building we had no features like that). Not sure we want to drop this V={1,2} part, though :D

No please don't add more verbose levels :) As for V=2 it's up to you, if you think it has value (I truly have no idea; I guess for CI logs it's often nice to have more things logged in general).

Reply:
I don't think it hurts but at the same time, it's not super useful. Let me have a look at it (maybe we can sync offline)

Reply:
ok, so we went back to the initial approach. V=1 only this time, anything other than 1 defaults to quiet.

Reply:
That works for me. The original comment was just about explicitly listing `1` as the expected value in the help.

Reply:
Had a look at the latest version, `V=2` was removed but `--verbose` is added in `CARGO_OPTS`. Isn't that making a case for supporting `V=2`; if that is useful in a common use case?

Sorry if we're going circles on this... :(

Reply:
> Had a look at the latest version, `V=2` was removed but `--verbose` is added in `CARGO_OPTS` in the CI. Isn't that making a case for supporting `V=2`; if that is useful in a common use case?
> 
> Sorry if we're going circles on this... :(

that was the "use case" I went/proposed that logic (and mostly because there weren't others really useful).
The point was that we used to build with a simple `cargo build --verbose` so I remapped this with the makefile logic.
I don't know, V=2 used in that way is not that useful, so I really have no strong preference on this.

Reply:
Let's keep it this way then, we can add it back in the future if we every see the need.